lib/kolekti/memory_persistence_strategy.rb

```diff
@@ -40,5 +44,30 @@ def create_related_hotspot_metric_results(metric_configuration, results)
 
       related_hotspot_metric_results << related_results
     end
+
+    private
+
+    def metric_already_in_use?(module_name, metric_configuration)
+      @tree_metric_results.any? do |metric|
+        has_same_name?(metric, module_name) and uses_same_configuration?(metric, metric_configuration)
+      end
+    end
+
+    def has_same_name?(metric, module_name)
+      metric[:module_name] == module_name
+    end
+
+    def uses_same_configuration?(metric, metric_configuration)
+      metric[:metric_configuration] == metric_configuration
+    end
+
   end
+
+  class AlreadyTakenModuleException < Exception
+    def initialize(module_name, metric)
+      reason = "The module #{module_name} or the metric #{metric} is already being used"
```

Comment:
What do you think about setting the message to: `The module #{module_name} already has a result stored for the metric #{metric}.`?